homeassistant/components/airios/config_flow.py

```diff
@@ -0,0 +1,629 @@
+from homeassistant.config_entries import (
+    SOURCE_RECONFIGURE,
+    SOURCE_USER,
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    ConfigSubentryFlow,
+    OptionsFlow,
+    SubentryFlowResult,
+)
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_TYPE,
+)
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_BRIDGE_RF_ADDRESS,
+    CONF_DEFAULT_HOST,
+    CONF_DEFAULT_NETWORK_MODBUS_ADDRESS,
+    CONF_DEFAULT_PORT,
+    CONF_DEFAULT_SERIAL_MODBUS_ADDRESS,
+    CONF_RF_ADDRESS,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SUPPORTED_ACCESSORIES,
+    SUPPORTED_UNITS,
+    BridgeType,
+)
+from .coordinator import AiriosDataUpdateCoordinator
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AiriosConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Airios."""
+
+    VERSION = 1
+
+    _reconfigure_data: MappingProxyType[str, Any]
+    _conf_modbus_address: int
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["serial", "network"],
+        )
+
+    async def _finish(self, entry_data: dict[str, Any]):
+        bridge_rf_address = entry_data[CONF_BRIDGE_RF_ADDRESS]
+        if self.source == SOURCE_USER:
+            return self.async_create_entry(
+                title=f"Airios RF bridge ({bridge_rf_address:06X})",
+                data=entry_data,
+            )
+        if self.source == SOURCE_RECONFIGURE:
+            return self.async_update_reload_and_abort(
+                self._get_reconfigure_entry(),
+                data=entry_data,
+            )
+        raise HomeAssistantError(f"Unknown config flow source {self.source}")
+
+    async def async_step_network(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when setting up a network bridge."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input[CONF_PORT]
+            modbus_address = user_input[CONF_ADDRESS]
+            try:
+                data = await self._async_validate_bridge_network(
+                    host=host, port=port, modbus_address=modbus_address
+                )
+            except UnexpectedProductId:
+                errors["base"] = "unexpected_product_id"
+            except AiriosException:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self._finish(data)
+
+        conf_host = CONF_DEFAULT_HOST
+        conf_port = CONF_DEFAULT_PORT
+        conf_modbus_address = CONF_DEFAULT_NETWORK_MODBUS_ADDRESS
```

Comment:
What's an address?

Reply:
The integration (and pyairios library) only supports the RS485 bridge. But the integration supports Modbus TCP in case a Modbus gateway (RTU <--> TCP) is used.  The Modbus address is the Modbus unit address of the bridge itself, 207 by default for the RS485 bridge.